.github/workflows/rust.yml

```diff
@@ -342,3 +342,67 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  test-examples:
+    name: Integration test ${{ matrix.name }}
+    runs-on: ${{ matrix.os }}
+    needs: build
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - { name: "Linux-x86_64",       target: x86_64-unknown-linux-musl,  os: ubuntu-20.04 }
+#          - { name: "Linux-aarch64",      target: aarch64-unknown-linux-musl, os: ubuntu-latest } Doesn't work yet
+
+          - { name: "macOS-x86_64",       target: x86_64-apple-darwin,        os: macOS-latest }
+          - { name: "macOS-aarch64",      target: aarch64-apple-darwin,       os: macOS-14 } # macOS-14 is the ARM chipset
+
+          - { name: "Windows-x86_64",     target: x86_64-pc-windows-msvc,     os: windows-latest , extension: .exe}
+    steps:
+      - name: checkout repo
+        uses: actions/checkout@v4
+      - name: Download binary from build
+        uses: actions/download-artifact@v4
+        with:
+          name: pixi-${{ matrix.target }}${{ matrix.extension }}
+          path: pixi_bin
+      - name: Debug
+        run: |
+          pwd
+      - name: Setup unix binary, add to github path
+        if: matrix.os != 'windows-latest'
+        run: |
+          mv pixi_bin/pixi-${{ matrix.target }} pixi_bin/pixi
+          chmod a+x pixi_bin/pixi
+          echo "$(pwd)/pixi_bin" >> $GITHUB_PATH
+      - name: Create Directory and Move Executable
+        if: matrix.os == 'windows-latest' && matrix.target == 'x86_64-pc-windows-msvc'
+        run: |
+          New-Item -ItemType Directory -Force -Path "D:\.pixi"
+          Move-Item -Path "pixi_bin/pixi-${{ matrix.target }}${{ matrix.extension }}" -Destination "D:\.pixi\pixi.exe"
+        shell: pwsh
+      - name: Add to PATH
+        if: matrix.os == 'windows-latest' && matrix.target == 'x86_64-pc-windows-msvc'
+        run: echo "D:\.pixi" | Out-File -Append -Encoding utf8 -FilePath $env:GITHUB_PATH
+        shell: pwsh
+      - name: Verify and Use Executable
+        if: matrix.os == 'windows-latest' && matrix.target == 'x86_64-pc-windows-msvc'
+        run: |
+          echo "Current PATH: $env:PATH"
+          pixi --version
+        shell: pwsh
+      - name: Help
+        run: pixi --help
+      - name: Info
+        run: pixi info
+      - name: Install pixi
+        run: pixi install -v
+      - name: Test examples/polarify
+        run: pixi run -v --manifest-path examples/polarify/pixi.toml test
+      - name: Test examples/pypi
+        # Skip windows until this error is solved: `× could not get artifact: Access is denied. (os error 5)`
+        if: matrix.os != 'windows-latest'
+        run: pixi run -v --manifest-path examples/pypi/pixi.toml test
+      - name: Test examples/solve-groups
+        run: |
+          pixi run -v --manifest-path examples/solve-groups/pixi.toml -e min-py38 test
+          pixi run -v --manifest-path examples/solve-groups/pixi.toml -e max-py310 test
```

Comment:
Wouldnt it make sense to just add a bash/nushell script to do this? That would also make it _much_ easier to run locally.

Reply:
Oh that is a good idea, as in put all the run scripts in a `test_examples.sh` ?

Reply:
Bash is luckily cross platform in actions. Nushell would have needed to be installed and I hate pwsh